doc/source/user_guide/categorical.rst

```diff
@@ -244,7 +244,7 @@ Equality semantics
 ~~~~~~~~~~~~~~~~~~
 
 Two instances of :class:`~pandas.api.types.CategoricalDtype` compare equal
-whenever they have the same categories and order. When comparing two
+whenever they have the same categories, order and dtypes. When comparing two
```

Comment:
If the categories do not have the same dtype, they are not the same. So I think the original sentence was better, however it would be helpful to add this to clarify. How about adding something like:

    Note that categories with different dtypes are not the same.